### PR TITLE
Provide a way for plugins to be written for slim 

### PR DESCRIPTION
**Pull request by m-mujica**
ue9rway

### CODE REVIEW

**Comment by m-mujica** [comment]
@matthewp this is a naive implementation but it works, I was able to get [steal-css/slim](https://github.com/stealjs/steal-css/blob/slim/slim.js) to be included in the build

ue9rway


I should be able to replicate the [build](https://github.com/m-mujica/slim-css/tree/master/out/bundles/slim-css) in my example tomorrow, need to get the config in place and the conditional in the require

**Comment by matthewp** [comment]
👍 looks good to me.

**Comment by m-mujica** [comment]
@matthewp I did quite a few changes here, mind taking a look again?

File: lib/stream/load_optimized_plugins.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		var pluginBuilder = node.value && node.value.pluginBuilder;
+
+		if (node.isPlugin && pluginBuilder) {
+			return loader
+				.normalize(pluginBuilder)
+				.then(function(name) {
+					return loader.locate({ name: name, metadata: {} });
+				})
+				.then(function(address) {
+					return Promise.all([address, readFile(address.replace("file:", ""))]);
```
Comment (matthewp) [inline lib/stream/load_optimized_plugins.js]: Wouldn't it be better to use `loader.fetch` here?

File: lib/stream/load_optimized_plugins.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+						return Promise.all([
+							name,
+							loader.locate({ name: name, metadata: {} })
+						]);
+					})
+					// [ name, address ]
+					.then(function(data) {
+						return Promise.all([
+							data[1],
+							loader.fetch({ name: data[0], address: data[1], metadata: {} })
```
Comment (m-mujica) [inline lib/stream/load_optimized_plugins.js]: @matthewp done!

Comment (matthewp) [inline lib/stream/load_optimized_plugins.js]: 👍 merge then

**Pull request merged**